### PR TITLE
Resolve Linting after Poetry Changes

### LINKED ISSUE #297: Reduce Pylint ignores after Poetry Changes
Pylint after the introduction of poetry and removal of pylintrc file that disabled many warning has too long a list in pyproject.toml.

This PR should dramatically reduce this list.

### PR DESCRIPTION
**Pull request by keithrfung**
### Issue

Fixes #297 

### Description
This resolves a considerable amount of pylint issues that were previously ignored. With the switch to poetry these were highlighted as being quite a long list. This takes it down as much as possible so other contributors can begin helping. 

### Testing
`make auto-lint`

### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] 🤔 **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] ✅ **DO** check open PR's to avoid duplicates.
- [x] ✅ **DO** keep pull requests small so they can be easily reviewed.
- [x] ✅ **DO** build locally before pushing.
- [x] ✅ **DO** make sure tests pass.
- [x] ✅ **DO** make sure any new changes are documented.
- [x] ✅ **DO** make sure not to introduce any compiler warnings.
- [x] ❌**AVOID** breaking the continuous integration build.
- [x] ❌**AVOID** making significant changes to the overall architecture.

💚Thank you!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -62,58 +62,27 @@ typish = '*'
 [tool.black]
 target-version = ['py38']
 
+[tool.pylint.'FORMAT']
+max-line-length = 120
+
 # FIXME: Pylint should not require this many exceptions
 [tool.pylint.'MESSAGES CONTROL']
 disable = '''
```
Comment (keithrfung) [inline pyproject.toml:70]: This is the import section to look at. There are a couple that are ignored in code but these places are typically exceptions.

File: pyproject.toml [hunk 2]
```diff
@@ -62,58 +62,27 @@ typish = '*'
 [tool.black]
 target-version = ['py38']
 
+[tool.pylint.'FORMAT']
+max-line-length = 120
```
Comment (keithrfung) [inline pyproject.toml:66]: Ideally we should get this down to `100` which is the default

Comment (AddressXception) [reply pyproject.toml:66]: im in favor of keeping it at 120.  It's the 21st century and we all have big screens

File: pyproject.toml [hunk 3]
```diff
@@ -62,58 +62,27 @@ typish = '*'
...
-  no-else-return,
-  no-name-in-module,
-  no-self-use,
   no-value-for-parameter,
-  protected-access,
   redefined-builtin,
-  redefined-outer-name,
-  super-init-not-called,
-  super-with-arguments,
   too-few-public-methods,
```
Comment (AddressXception) [inline pyproject.toml:78]: lol how is this even a thing

File: pyproject.toml [hunk 4]
```diff
@@ -62,58 +62,27 @@ typish = '*'
...
   too-many-arguments,
   too-many-branches,
   too-many-function-args,
   too-many-lines,
   too-many-locals,
-  too-many-instance-attributes,
   too-many-nested-blocks,
-  too-many-public-methods,
```
Comment (AddressXception) [inline pyproject.toml:105]: also here, why not ignore this?  how many is too many public methods?

**Review by AddressXception — APPROVED** [review]

**Pull request merged**